webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonQueryObjects.scala

```diff
@@ -102,7 +102,10 @@ object JsonQueryObjects {
   ) {
 
     def update(ruleCategory: RuleCategory) = {
-      ruleCategory.using(this).ignoreRedundantPatcherFields.patch
+      ruleCategory.copy(
+        name = name.getOrElse(ruleCategory.name),
+        description = description.getOrElse(ruleCategory.description)
```

Comment:
I think it miss the parent

Reply:
Perhaps also check why there is an ID here, because I'm not sure what it means to change the id. Appart (mayhaps) in the clone case ?

Reply:
'parent' was never used as a patched value (the `RuleCategory` doesn't have a 'parent' attribute, and this is why the `ignoreRedundantPatcherFields` was required here)

Nor should the `id` be replaced because the `id` of both are always the same when doing the 'update'